elements/pl-external-grader-variables/pl-external-grader-variables.py

```diff
@@ -1,34 +1,90 @@
 import chevron
-import lxml.html
+import lxml
 import prairielearn as pl
 
-PARAMS_NAME_DEFAULT = None
+EMPTY_DEFAULT = False
 
 
-def prepare(element_html, element_index, data):
+def prepare(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
-    pl.check_attribs(element, ["params-name"], [])
+    pl.check_attribs(element, ["params-name"], ["empty"])
 
-    params_name = pl.get_string_attrib(element, "params-name", PARAMS_NAME_DEFAULT)
-    if params_name is None:
-        raise Exception('Attribute "params-name" is not defined.')
-    if params_name not in data["params"]:
-        raise Exception(
-            f"Variable name {params_name} does not exist in data['params']."
-        )
+    params_name = pl.get_string_attrib(element, "params-name")
 
+    # Get frontend variables
```

Comment:
I get what you're going for with "frontend", but we don't use that terminology elsewhere (I'm trying hard to avoid introducing new terminology for things):

```suggestion
    # Get any variables defined in HTML
```